pkg/objectstate/rte/rte.go

```diff
@@ -358,8 +412,37 @@ func FromClient(ctx context.Context, cli client.Client, plat platform.Platform,
 
 	// should have the amount of resources equals to the amount of node groups
 	for _, tree := range trees {
-		for _, mcp := range tree.MachineConfigPools {
-			generatedName := objectnames.GetComponentName(instance.Name, mcp.Name)
+		if plat == platform.OpenShift {
+			for _, mcp := range tree.MachineConfigPools {
+				generatedName := objectnames.GetComponentName(instance.Name, mcp.Name)
+				key := client.ObjectKey{
+					Name:      generatedName,
+					Namespace: namespace,
+				}
+				ds := &appsv1.DaemonSet{}
+				dsm := daemonSetManifest{}
+				if dsm.daemonSetError = cli.Get(ctx, key, ds); dsm.daemonSetError == nil {
+					dsm.daemonSet = ds
+				}
+				ret.daemonSets[generatedName] = dsm
+
+				if plat == platform.OpenShift {
```

Comment:
This flow is already in `OpenShift` case, no need to check again

Reply:
right, that's a mistake, fixed. thanks